app/models/concerns/media_archiver.rb

```diff
@@ -93,20 +93,47 @@ def handle_archiving_exceptions(archiver, params)
       begin
         ApiKey.current = ApiKey.find_by(id: params.dig(:key_id))
         yield
+      rescue Pender::Exception::RetryLimitHit => error
+        handle_rate_limit_error(archiver, params, error)
```

Comment:
Is it right that it's calling the `handle_rate_limit` callback for a "retry limit hit" exception?

Reply:
This is now updated to include a dedicated method to handle all `limit` errors